tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/PowSetCtor.scala

```diff
@@ -40,8 +41,13 @@ class PowSetCtor(rewriter: SymbStateRewriter) {
 
     rewriter.solverContext.log("; %s(%s) {".format(getClass.getSimpleName, state.ex))
     val powSetSize = BigInt(1) << elems.size
-    if (powSetSize >= Limits.POWSET_MAX_SIZE) {
-      throw new RewriterException(s"Attempted to expand a powerset of size 2^${elems.size}", state.ex)
+    if (powSetSize > Limits.POWSET_MAX_SIZE) {
+      logger.error("This error typically occurs when one enumerates all subsets:"
+        + " \\A S \\in SUBSET T: P or \\E S \\in SUBSET T: P")
+      logger.error("Try to decrease the cardinality of the base set T, or avoid enumeration.")
+      val msg =
+        s"Attempted to expand SUBSET of size 2^${elems.size}, whereas the built-in limit is ${Limits.POWSET_MAX_SIZE}"
```

Comment:
Printing 1048576 here may look confusing to the user.

Why not store `Limits.POWSET_MAX_SIZE` as the number of elements?
Then you can directly compare `elems.size` above and print it here as a power of 2.

Reply:
That's true. I will fix that :)